npm/pkg/controlplane/controllers/v1/nameSpaceController.go

```diff
@@ -348,6 +360,8 @@ func (nsc *NamespaceController) syncAddNameSpace(nsObj *corev1.Namespace) error
 
 // syncUpdateNameSpace handles updating namespace in ipset.
 func (nsc *NamespaceController) syncUpdateNameSpace(newNsObj *corev1.Namespace) error {
+	timer := metrics.StartNewTimer()
```

Comment:
Instead of dealing these metrics here, we can just put then at the start of syncAddNS and cleanupDeleteNS, that will keep the code fairly simple ?